test/integration/fake_upstream.cc

```diff
@@ -585,28 +545,29 @@ FakeUpstream::waitForHttpConnection(Event::Dispatcher& client_dispatcher,
     return AssertionFailure() << "No upstreams configured.";
   }
   Event::TestTimeSystem& time_system = upstreams[0]->timeSystem();
-  auto end_time = time_system.monotonicTime() + timeout;
-  while (time_system.monotonicTime() < end_time) {
+  auto end_time = time_system.realMonotonicTimeDoNotUseWithoutScrutiny() + timeout;
```

Comment:
ditto for calling the system function directly with the nolint escape (though this is ok too)